File: dosen.py
from human import Human
# Kelas Dosen sebagai child dari kelas Human
class Dosen(Human) :

    #access modifier private
    __nip ="";
    __nama ="";
    __jenis_kelamin ='';
    __fakultas ="";
    __pend_terakhir ="";
    __keahlian ="";
    
    # constructor
    def __init__(self,  nama = "", nip = "", jenis_kelamin = '', fakultas = "", pend_terakhir = "", keahlian = "" ):
        self.__nip = nip
        self.__nama = nama
        self.__jenis_kelamin = jenis_kelamin
        self.__fakultas = fakultas
        self.__pend_terakhir = pend_terakhir
        self.__keahlian = keahlian
        self.univ = None

    # getter and setter
    def get_nip(self):
        return self.__nip
    
    def set_nip(self, nip):
        self.__nip = nip

    def get_nama(self):
        return self.__nama
    
    def set_nama(self, nama):
        self.__nama = nama

    def get_jenis_kelamin(self):
        return self.__jenis_kelamin
    
    def set_jenis_kelamin(self, jenis_kelamin):
        self.__jenis_kelamin = jenis_kelamin

    def get_fakultas(self):
        return self.__fakultas
    
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

    def get_pend_terakhir(self):
        return self.__pend_terakhir
    
    def set_pend_terakhir(self, pend_terakhir):
        self.__pend_terakhir = pend_terakhir

    def get_keahlian(self):
        return self.__keahlian
    
    def set_keahlian(self, keahlian):
        self.__keahlian = keahlian

File: cpp/dosen.cpp
//import library
#include <bits/stdc++.h>
#include "human.cpp"
//using standard namespace
using namespace std;
//Human sebagai anak yang diwariskan dari Mahasiswa
class Dosen : public Human
{
    //access modifier private
    private:
        string nip;
        string nama;
        string jenis_kelamin;
        string fakultas;
        string pend_terakhir;
        string keahlian;

    //access modifier publik    
    public:
        string univ;
        //constructor
        Dosen()
        {
            this->nip = " ";
            this->nama = " ";
            this->jenis_kelamin = " ";
            this->fakultas = " ";
            this->pend_terakhir = " ";
            this->keahlian = " ";
            this->univ= " ";
        }

        //penggunaan getter dan setter
        string get_nip()
        {
            return this->nip;
        }

        void set_nip(string nip)
        {
            this->nip = nip;
        }
        
        string get_nama()
        {
            return this->nama;
        }

        void set_nama(string nama)
        {
            this->nama = nama;
        }

        char get_jenis_kelamin()
        {
            return this->jenis_kelamin;
        }

        void set_jenis_kelamin(string jenis_kelamin)
        {
            this->jenis_kelamin = jenis_kelamin;
        }

        string get_fakultas()
        {
            return this->fakultas;
        }

        void set_fakultas(string fakultas)
        {
            this->fakultas = fakultas;
        }

        string get_pend_terakhir()
        {
            return this->pend_terakhir;
        }

        void set_pend_terakhir(string pend_terakhir)
        {
            this->pend_terakhir = pend_terakhir;
        }

        string get_keahlian()
        {
            return this->keahlian;
        }

        void set_keahlian(string keahlian)
        {
            this->keahlian = keahlian;
        }

        //destructor
        ~Dosen()
        {

        }
};